lib/preprocessing/shared.ex

```diff
@@ -132,8 +132,22 @@ defmodule Mighty.Preprocessing.Shared do
     ]
   ]
 
+  bm25_schema_opts = [
+    term_saturation_factor: [
+      type: :float,
+      default: 1.2,
+      doc: "Controls non-linear term frequency normalization (saturation) in BM25."
+    ],
+    length_normalization_factor: [
+      type: {:custom, __MODULE__, :validate_float_0_to_1, []},
```

Comment:
You can find it used in other validations in the schemas I have, but you can you an interval sigil here from https://github.com/acalejos/exterval

This would look like:

`type: {:in, ~i<[0,1]>}